spring/boot2-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/ReactiveHealthEndpointWebExtensionUtil.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.spring.actuate;
+
+import static com.linecorp.armeria.spring.actuate.WebOperationService.handleResult;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.reactivestreams.Publisher;
+import org.springframework.boot.actuate.endpoint.InvalidEndpointRequestException;
+
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+final class ReactiveHealthEndpointWebExtensionUtil {
+
+    static HttpResponse handleMaybeReactorResult(SimpleHttpCodeStatusMapper statusMapper,
+                                                 ServiceRequestContext ctx, Object result,
+                                                 HttpMethod method) throws Throwable {
+        if (result instanceof Flux) {
+            return handlePublisher(statusMapper, ctx, ((Flux<?>) result).collectList(), method);
+        }
+
+        if (result instanceof Publisher) {
+            return handlePublisher(statusMapper, ctx, (Publisher<?>) result, method);
+        }
+
+        return handleResult(statusMapper, ctx, result, method);
+    }
+
+    private static HttpResponse handlePublisher(SimpleHttpCodeStatusMapper statusMapper,
+                                                ServiceRequestContext ctx, Publisher<?> result,
+                                                HttpMethod method) {
+        final CompletableFuture<HttpResponse> future = new CompletableFuture<>();
+        Mono.from(result).subscribe(result0 -> {
```

Comment:
This logic is copied from [here](https://github.com/spring-projects/spring-boot/blob/main/spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/web/reactive/AbstractWebFluxEndpointHandlerMapping.java#L373-L382)

Reply:
Not sure we really need backpressure here. But `.subscribe()` seems to break backpressure. 
Should we use `.map` and `StreamMessage.concat()` to connect the stream changes?
For example:
```java
Mono<HttpResponse> result = Mono.from(result)
                                .map(result0 ->  ... );
StreamMessage<HttpObject> concatenated = StreamMessage.concat(result);
return HttpResponse.of(concatenated);
```

Reply:
> Not sure we really need backpressure here.

We need that. 😄 Thanks for catching this. 👍 👍